python/pylibcudf/pylibcudf/column.pyx

```diff
@@ -321,6 +338,80 @@ cdef class Column:
             []
         )
 
+    @singledispatchmethod
+    @classmethod
+    def from_any(cls, obj):
```

Comment:
Suggestion: Maybe we can start with an `from_ndarray` method that accepts `numpy` and `cupy` arrays, and then build up an `from_any` when we build up more ingestion methods (e.g. `from_pyobject`, `from_pandas`, etc)

Reply:
Would we eventually remove `from_ndarray` in favor of`from_any`? If so, I'd rather keep `from_any` since its ultimately what we want. And because we're planning on adding more ingestion methods during 25.06 release.

If we plan to keep it, then I'll change `from_any` to `from_ndarray`. I wanted to add a stable API sooner rather than later for Curator's use case. They need to be able create a list column from a cupy array. It should be public and stable so they can keep doing something like the following without it breaking between cudf releases.
```python
cudf.Series.from_pylibcudf(
    plc.Column.from_any(
        cupy_array
    )
)
```

*Also forgot to add a doc string, will add that too.

Reply:
Ah OK I do see now that @vyasr proposed in #15132 that a `from_any` method would just dispatch to another public method e.g. `from_ndarray` (for example), by `singledispatch`. So we would have both a public `from_any` and `from_ndarray` on the column method in this PR.

Reply:
Yes, but I'll start with `from_ndarray` in this PR and add `from_any` in a follow-up PR.